src/stream.c

```diff
@@ -276,31 +273,30 @@ void avifRWStreamFinishWrite(avifRWStream * stream)
     }
 }
 
-void avifRWStreamWrite(avifRWStream * stream, const void * data, size_t size)
+avifResult avifRWStreamWrite(avifRWStream * stream, const void * data, size_t size)
 {
-    if (!size) {
-        return;
+    if (size) {
+        AVIF_CHECKRES(makeRoom(stream, size));
+        memcpy(stream->raw->data + stream->offset, data, size);
+        stream->offset += size;
     }
-
-    makeRoom(stream, size);
-    memcpy(stream->raw->data + stream->offset, data, size);
-    stream->offset += size;
+    return AVIF_RESULT_OK;
 }
 
-void avifRWStreamWriteChars(avifRWStream * stream, const char * chars, size_t size)
+avifResult avifRWStreamWriteChars(avifRWStream * stream, const char * chars, size_t size)
 {
-    avifRWStreamWrite(stream, chars, size);
+    return avifRWStreamWrite(stream, chars, size);
 }
 
-avifBoxMarker avifRWStreamWriteFullBox(avifRWStream * stream, const char * type, size_t contentSize, int version, uint32_t flags)
+avifResult avifRWStreamWriteFullBox(avifRWStream * stream, const char * type, size_t contentSize, int version, uint32_t flags, avifBoxMarker * marker)
 {
-    avifBoxMarker marker = stream->offset;
+    *marker = stream->offset;
```

Comment:
[This issue is moot if we remove the `marker` output parameter.]

We can allow `marker` to be null if the caller doesn't need to know the stream's offset before the full box. This is the case when `contentSize` is not `AVIF_BOX_SIZE_TBD`.

I annotated the two places where this could be useful.